pkg/ruler/base/api_test.go

```diff
@@ -163,6 +164,53 @@ func TestRuler_alerts(t *testing.T) {
 	require.Equal(t, string(expectedResponse), string(body))
 }
 
+func TestRuler_GetRulesLabelFilter(t *testing.T) {
+	cfg := defaultRulerConfig(t, newMockRuleStore(mockRules))
+
+	r := newTestRuler(t, cfg)
+	defer r.StopAsync()
+
+	a := NewAPI(r, r.store, log.NewNopLogger())
+
+	tc := []struct {
+		name     string
+		URLQuery string
+		output   string
+		err      error
+	}{
+		{
+			name:   "query with no filters",
+			output: "test:\n    - name: group1\n      interval: 1m\n      rules:\n        - record: UP_RULE\n          expr: up\n        - alert: UP_ALERT\n          expr: up < 1\n          labels:\n            foo: bar\n        - alert: DOWN_ALERT\n          expr: down < 1\n          labels:\n            namespace: delta\n",
```

Comment:
This is hard to read, and will be hard to edit too.
Can we rather parse the responses and deep compare the result to a struct?

Reply:
Sure we can, that's what I did initially, but it made the test lengthier because of the steps it took to get the right data structure, but maybe my implementation wasn't the best. I'll re implement it so you can tell me if that's what you had in mind.